basic/src/test/java/org/jboss/hal/testsuite/test/configuration/ee/EETestCaseAbstract.java

```diff
@@ -0,0 +1,147 @@
+package org.jboss.hal.testsuite.test.configuration.ee;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.jboss.arquillian.drone.api.annotation.Drone;
+import org.jboss.arquillian.graphene.Graphene;
+import org.jboss.arquillian.graphene.findby.ByJQuery;
+import org.jboss.arquillian.graphene.page.Page;
+import org.jboss.hal.testsuite.cli.CliClientFactory;
+import org.jboss.hal.testsuite.cli.DomainManager;
+import org.jboss.hal.testsuite.dmr.AddressTemplate;
+import org.jboss.hal.testsuite.dmr.DefaultContext;
+import org.jboss.hal.testsuite.dmr.Dispatcher;
+import org.jboss.hal.testsuite.dmr.ResourceAddress;
+import org.jboss.hal.testsuite.dmr.ResourceVerifier;
+import org.jboss.hal.testsuite.dmr.StatementContext;
+import org.jboss.hal.testsuite.finder.Application;
+import org.jboss.hal.testsuite.finder.FinderNames;
+import org.jboss.hal.testsuite.finder.FinderNavigation;
+import org.jboss.hal.testsuite.fragment.ConfigFragment;
+import org.jboss.hal.testsuite.page.config.ConfigurationPage;
+import org.jboss.hal.testsuite.page.config.DomainConfigEntryPoint;
+import org.jboss.hal.testsuite.page.config.StandaloneConfigEntryPoint;
+import org.jboss.hal.testsuite.util.ConfigUtils;
+import org.jboss.hal.testsuite.util.Console;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * @author Jan Kasik
+ *         Created on 8.9.15.
+ */
+public class EETestCaseAbstract {
+
+    @Drone
+    protected WebDriver browser;
+
+    @Page
+    protected ConfigurationPage page;
+
+    protected Dispatcher dispatcher;
+    protected ResourceVerifier verifier;
+    protected ResourceAddress eeAddress;
+    private StatementContext context;
+
+    @Before
+    public void mainBefore() {
+        context  = new DefaultContext();
+        dispatcher = new Dispatcher();
+        verifier = new ResourceVerifier(dispatcher);
+        eeAddress =  AddressTemplate.of("{default.profile}/subsystem=ee").resolve(context);
+    }
+
+    @After
+    public void mainAfter() {
+        reloadIfRequiredAndWaitForRunning();
+        dispatcher.close();
+    }
+
+    protected void navigateToEEServices() {
+        FinderNavigation navigation;
```

Comment:
Please create the common interface for navigation to an application and move this to the implementation of this interface as we discussed f2f.